### DISCUSSION TITLE
Weasis doesn't show all series before downloading the images inside them

### DISCUSSION THREAD

**Discussion by nezaba** [nroduit/Weasis#378] [Q&A] [answered]
Hey, my problem is that weasis doesn't show all series before downloading the images inside them. It downloads the first series, all the images inside it, then continues with the others series in the same way. I tried changing the config **"download.concurrent.series"** and the option **"Download all series immediately"** is ticked but it keeps the same way. Reading the shortcuts documentation, it says  **"Downloading Priorities: Selecting a thumbnail gives the best priority to download."** so i guess its possible to download all the series before the images inside them to manage the priority. Anyone else's having this problem? thanks in advance

**Comment by nroduit** [comment]
As there are many methods to open images, it is necessary to explain the context of use because not all methods work in the same way.

Does it work well from the [demo site](https://nroduit.github.io/en/demo)?

**Reply by nezaba** [reply]
Hey, thanks for the response. Im using a Synapse pacs so i import the images with DICOM query/retrieve. When i use an old synapse dicom viewer with the same pacs, it downloads all series at once but weasis does it in sequence. I tried the demo site launching the **"DICOM Key Object Selection (KO)"** option and it works like it is supposed to, downloads all series and it let me manage the priority. Is there any config to make it work the same way with the query/retrieve method?

**Comment by nroduit** [comment, marked as answer]
No, unfortunately this is not possible with the DICOM Q/R (DIMSE) protocol. It is possible with the DICOMWeb protocol or by [integrating an intermediate component](https://nroduit.github.io/en/basics/customize/integration) (weasis-pacs-connector, the PACS must support the WADO-URI protocol).
